src/routes/api/token.js

```diff
@@ -0,0 +1,39 @@
+import { connect } from 'hyper-connect'
+import { PublicKey } from "@solana/web3.js";
+
+const hyper = connect(import.meta.env.VITE_HYPER_TOKEN)
+
+// test to see if wallet address is valid
+function validateSolAddress(address) {
+  try {
+    let pubkey = new PublicKey(address);
+    let isSolana = PublicKey.isOnCurve(pubkey.toBuffer());
+    return isSolana;
+  } catch (e) {
+    return false;
+  }
+}
+
+export async function post({ body }) {
+  // check and make sure wallet is valid
+  if (validateSolAddress(body.address)) {
+    // set default response
+    let result = { ok: true }
+    // if valid, make sure it has not already been added.
+    const doc = await hyper.data.get(body.address)
```

Comment:
Definitely an edge case, but maybe the address should be `trimmed` to prevent someone prepending or appending spaces to their address, either by accident or on purpose.

Reply:
Done